include/dlaf/cublas_tile.h

```diff
@@ -0,0 +1,202 @@
+//
+// Distributed Linear Algebra with Future (DLAF)
+//
+// Copyright (c) 2018-2020, ETH Zurich
+// All rights reserved.
+//
+// Please, refer to the LICENSE file in the root directory.
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+#pragma once
+
+/// @file
+
+#ifdef DLAF_WITH_CUDA
+
+#include <cublas_v2.h>
+#include <blas.hh>
+
+#include "dlaf/cublas/error.h"
+#include "dlaf/matrix/tile.h"
+#include "dlaf/types.h"
+#include "dlaf/util_cublas.h"
+
+// TODO: Which namespace?
+namespace dlaf {
+namespace internal {
+namespace gpu {
+
+template <typename T>
+struct CublasTrsm;
+
+template <>
+struct CublasTrsm<float> {
+  template <typename... Args>
+  static void call(Args&&... args) {
+    DLAF_CUBLAS_CALL(cublasStrsm(std::forward<Args>(args)...));
+  }
+};
+
+template <>
+struct CublasTrsm<double> {
+  template <typename... Args>
+  static void call(Args&&... args) {
+    DLAF_CUBLAS_CALL(cublasDtrsm(std::forward<Args>(args)...));
+  }
+};
+
+template <>
+struct CublasTrsm<std::complex<float>> {
+  template <typename... Args>
+  static void call(Args&&... args) {
+    DLAF_CUBLAS_CALL(cublasCtrsm(std::forward<Args>(args)...));
+  }
+};
+
+template <>
+struct CublasTrsm<std::complex<double>> {
+  template <typename... Args>
+  static void call(Args&&... args) {
+    DLAF_CUBLAS_CALL(cublasZtrsm(std::forward<Args>(args)...));
+  }
+};
+
+template <class T>
+void cublas_trsm(cublasHandle_t handle, blas::Side side, blas::Uplo uplo, blas::Op op, blas::Diag diag,
+                 T alpha, const matrix::Tile<const T, Device::GPU>& a,
+                 const matrix::Tile<T, Device::GPU>& b) {
```

Comment:
A couple of weeks ago we added `const` to all arguments of LAPACK and BLAS wrappers (see #276). I actually don't know how cuBLAS works, sorry. But do you think it is worth marking them as `const` here too?

Reply:
Yeah, absolutely, thanks!

Reply:
Updated.